include/xtypes/UnionType.hpp

```diff
@@ -0,0 +1,1064 @@
+/*
+ * Copyright 2020, Proyectos y Sistemas de Mantenimiento SL (eProsima).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef EPROSIMA_XTYPES_UNION_TYPE_HPP_
+#define EPROSIMA_XTYPES_UNION_TYPE_HPP_
+
+#include <xtypes/AggregationType.hpp>
+#include <xtypes/AliasType.hpp>
+#include <xtypes/EnumerationType.hpp>
+
+#include <string>
+#include <map>
+#include <vector>
+#include <regex>
+#include <codecvt>
+
+namespace eprosima {
+namespace xtypes {
+
+class DynamicData;
+class ReadableDynamicDataRef;
+class WritableDynamicDataRef;
+
+static const std::string UNION_DISCRIMINATOR("discriminator");
+static const size_t DEFAULT_UNION_LABEL = SIZE_MAX;
```

Comment:
It will surely fail. Changed the label type to `int64_t`, and `DEFAULT_UNION_LABEL` is now `std::numeric_limits<int64_t>::max()`.
So the problematic values are now something like `0x0FFFFFFFFFFFFFFF` and `0x0FFFFFFFFFFFFFFE` which should be very strange for a user to use them.

Reply:
Added a check to verify that now the user can add a `-1` label.